code/game/machinery/computer/launchpad_control.dm

```diff
@@ -86,27 +86,13 @@
 	if(isnull(the_pad))
 		return
 
-	var/x_dest = x_pos.input_value
-	if(isnull(x_dest))
-		return
-
-	var/y_dest = y_pos.input_value
-	if(isnull(y_dest))
-		return
-
-	if(x_dest > the_pad.range || y_dest > the_pad.range)
-		why_fail.set_output("Cannot go that far! Current maximum reach: [the_pad.range]")
-		on_fail.set_output(COMPONENT_SIGNAL)
-		return
```

Comment:
From my own messing around with this back when I was coding this, not having this fail condition made making stuff with it really stressful, since you had no idea what the hell was the problem. Is there any way to keep it? Maybe replace ``x_dest`` and ``y_dest`` with ``x_pos.input_value`` ``y_pos.input_value``, and when it's triggered set the offset to ``null, null``, and then add an else statement that sets the offset correctly?

Reply:
It clamps the value, it's not really a big deal

Reply:
Also it's less user friendly to have your program fail when it really shouldn't or flag up an error every time for something that works.

Reply:
I'd rather have that than having the user be frustrated because they tried to teleport to x: 55 y:2 and it bringing them to somewhere different and they not knowing why

Reply:
I'll add the range information to the ui notices. Other than that, I don't think it should be flagging an error for something that 'll work.

Reply:
Not only that, this isn't a properly set up if statement because it doesn't check for the minimum range either.

Reply:
I didn't think about using the ui notices because I didn't know when they refreshed, I thought it was a one-off thing that only happened when the component was inserted into the circuit, but if you say that it's fine then it should be fine. And yeah I like that circuits are as robust as they can be, I just want the user to be able to figure out why it isn't working easily

Reply:
It updates whenever the `i` button is pressed